framework/src/eigensolver/monolith.c

```diff
@@ -0,0 +1,559 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#if LIBMESH_HAVE_SLEPC
+
+#include <slepc/private/epsimpl.h> /*I "slepceps.h" I*/
+#include <slepcblaslapack.h>
+#include <petscsnes.h>
+
+typedef struct {
```

Comment:
It is not necessary. I am going to add to one more parameter. When the parameter is turn to zero, the Newton will be the same as the free power iteration. And then we could gradually increase the parameter. This idea is better.